astropy/units/format/latex.py

```diff
@@ -52,41 +52,31 @@ def _format_unit_list(cls, units):
 
     @classmethod
     def to_string(cls, unit, inline=False):
-        latex_name = None
-        if hasattr(unit, "_format"):
-            latex_name = unit._format.get("latex")
-
-        if latex_name is not None:
-            s = latex_name
-        elif isinstance(unit, core.CompositeUnit):
-            if unit.scale == 1:
-                s = ""
+        if unit.scale == 1:
+            s = ""
+        else:
+            s = cls.format_exponential_notation(unit.scale)
+
+        if len(unit.bases):
+            if s:
+                s += r"\,"
+            if inline:
+                nominator = zip(unit.bases, unit.powers)
+                denominator = []
             else:
-                s = cls.format_exponential_notation(unit.scale)
-
-            if len(unit.bases):
-                if s:
-                    s += r"\,"
-                if inline:
-                    nominator = zip(unit.bases, unit.powers)
-                    denominator = []
-                else:
-                    nominator, denominator = utils.get_grouped_by_powers(
-                        unit.bases, unit.powers
-                    )
-                if len(denominator):
-                    if len(nominator):
-                        nominator = cls._format_unit_list(nominator)
-                    else:
-                        nominator = "1"
-                    denominator = cls._format_unit_list(denominator)
-                    s += rf"\frac{{{nominator}}}{{{denominator}}}"
-                else:
+                nominator, denominator = utils.get_grouped_by_powers(
+                    unit.bases, unit.powers
+                )
+            if len(denominator):
+                if len(nominator):
                     nominator = cls._format_unit_list(nominator)
-                    s += nominator
-
-        elif isinstance(unit, core.NamedUnit):
-            s = cls._latex_escape(unit.name)
+                else:
+                    nominator = "1"
+                denominator = cls._format_unit_list(denominator)
+                s += rf"\frac{{{nominator}}}{{{denominator}}}"
+            else:
+                nominator = cls._format_unit_list(nominator)
+                s += nominator
 
```

Comment:
As this is very similar with what is in `console.py` and `generic.py`, couldn't this also united? By having a `format_fraction(nominator, denominator)` method that takes the specifics of console and latex?
```Python
class Generic(Base): […]
    @classmethod
    def to_string(cls, unit, inline=False):
        if unit.scale == 1:
            s = ""
        else:
            s = cls.format_exponential_notation(unit.scale)

        if len(unit.bases):
            if s:
                s += " "
            if inline:
                nominator = zip(unit.bases, unit.powers)
                denominator = []
            else:
                nominator, denominator = utils.get_grouped_by_powers(
                    unit.bases, unit.powers
                )
            if len(denominator):
                if not len(nominator):
                    nominator = ["1"]
                s += cls._format_fraction(nominator, denominator)
            else:
                s += cls._format_unit_list(nominator)

        return s

    @classmethod
    def _format_fraction(nominator, denominator):
        if len(denominator == 1):
            return f"{cls._format_unit_list(nominator)} / {cls._format_unit_list(denominator)}"
        else:
            return f"{cls._format_unit_list(nominator)} / ({cls._format_unit_list(denominator)})"


class Latex(Generic): […]
    @classmethod
    def _format_fraction(nominator, denominator):
        return rf"\frac{{{cls._format_unit_list(nominator)}}}{{{cls._format_unit_list(denominator)}}}"

    @classmethod
    def to_string(cls, unit, inline=True):
        # properly wrap into LaTeX mathmode
        s = super().to_string(unit, inline)
        return rf"$\mathrm{{{s}}}$"


class Console(Generic): […]
    @classmethod
    def _format_fraction(nominator, denominator):
        nominator = cls._format_unit_list(denominator)
        denominator = cls._format_unit_list(denominator)
        fraclength = max(len(nominator), len(denominator))
        f = f"{{0:<{len(s)}s}}{{1:^{fraclength}s}}"

        return "\n".join(
            (
                f.format("", nominator),
                f.format(s, cls._line * fraclength),
                f.format("", denominator),
            )

    @classmethod
    def to_string(cls, unit, inline=True):
        return super().to_string(unit, inline)
```
This would concentrate all the handling of the different options (with/without de/nominator, inline) in a general re-usable `to_string() method and have the `inline` parameter more general.
Or does this get too abstract then?

Reply:
Hah, those were just my thoughts, and indeed the next step includes moving `to_string()` -- see #14416! (move to `Base`, since `console`, etc., shouldn't inherit the parsing form `generic`).

Aside, while so far (including #14416) the tests do not change, I have been thinking that perhaps, while we still can, we should rename `inline` -- it made sense for `latex` but I feel it is somewhat unclear for other formats, and it would also be nice to separate in-lline fractions from using negative powers. The best I could come up with so far is `fraction=False`, `fraction='small'` and `fraction='big'` (with `fraction=True` whatever makes most sense). Hopefully to extend to `fits` and `ogip`, where there is a wish to allow things like "Jy/beam" (see #13217).

Reply:
Agreed.